### DISCUSSION TITLE
Cannot deploy typst rendered passage

### DISCUSSION THREAD

**Discussion by hz-xiaxz** [Myriad-Dreamin/typst.ts#616] [Q&A] [answered]
Hello, thanks for creating this awesome repository! 
I'm using [hexo-renderer-typst](https://socket.dev/npm/package/hexo-renderer-typst) on my own blog, I find it successfully rendered locally with `hexo server`, but fail to view after `hexo deploy` on github pages. Is it currently not enabled or I just config something wrong? Looking forward to your kind help!

**Comment by Myriad-Dreamin** [comment]
what's the specific error? I guess you ran into the error mentioned [here](https://github.com/Myriad-Dreamin/typst.ts/issues/515#issuecomment-2054109236).

**Comment by hz-xiaxz** [comment]
Hi, this is the content label of my generated html
```html
<meta property="og:description" content="window.typstBindSvgDom &#x3D; function () {};
 window.captureStack &#x3D; function () {
   return undefined;
 };
 let appContainer &#x3D; document.currentScript &amp;&amp; document.currentScript.parentElement;
 
 const appE">
```

when I run `hexo server` on my local laptop it works,
![image](https://github.com/user-attachments/assets/15872926-9dfd-4596-8741-42c22acd0254)
![image](https://github.com/user-attachments/assets/5ca84cda-12af-4256-8467-4bd244c5104c)
However when I tried to upload to a [github page](https://hz-xiaxz.github.io/2024/12/13/example/) the typst rendered image disappears.

I wonder if it is because my CI didn't upload the generated svg rightly. If so, could you tell me where is the rendered svg stored?

**Comment by hz-xiaxz** [comment]
Or if the hexo-typst plugin depends on a typst environment on the server, so the github page won't work?

**Comment by Myriad-Dreamin** [comment, marked as answer]
I find an error in the console when opening your website. "https://hz-xiaxz.github.io/artifacts/typst/source/_posts/template.multi.sir.in 404 (Not Found)". The artifacts must be already uploaded to the correct place. https://github.com/hz-xiaxz/hz-xiaxz.github.io/tree/main/artifacts/typst/source/_posts. The provided relative URL may be wrong so you have to fix it.

**Reply by hz-xiaxz** [reply]
That's it, many thanks for your help!
